tempodb/compaction_block_selector.go

```diff
@@ -124,6 +126,7 @@ func (twbs *timeWindowBlockSelector) BlocksToCompact() ([]*backend.BlockMeta, st
 				stripe := twbs.entries[i : j+1]
 				if twbs.entries[i].group == twbs.entries[j].group &&
 					twbs.entries[i].meta.DataEncoding == twbs.entries[j].meta.DataEncoding &&
+					twbs.entries[i].meta.Version == twbs.entries[j].meta.Version && // update after parquet: only compact blocks of the same version
```

Comment:
it makes sense that we don't want to compact `v2` and `vparquet` but as we iterate on parquet will this condition still be true?

should we write a function `versionsCanCompact(v1 string, v2 string)` and just prevent v2 from compacting with any other v2? but allow future parquet versions to compact?

Reply:
Our plan here is to make `vParquet` tolerant of added/dropped columns as we go.  We think this should be decently straightforward.   When we introduce a change large enough where this is no longer true, we'll make it a new block Version instead of a data version, and compaction will keep them separated still.